src/core/test/index.js

```diff
@@ -127,10 +127,10 @@ const test = (fieldName, ...args) => {
         testFn,
         severity;
 
-    if (typeof args[0] === 'string') {
-        [statement, testFn, severity] = args;
-    } else if (isTestFn(args[0])) {
+    if (isTestFn(args[0])) {
         [testFn, severity] = args;
+    } else if (['string', 'object'].some((type) => typeof args[0] === type)) {
+        [statement, testFn, severity] = args;
```

Comment:
The order was flipped because Promise is an object and we don't want it to fall inside our condition so we check now if it's a Promise and then we do the other check we have.